src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UpdateInterpreterDetailsMidEventHandler.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.INTERPRETER_DETAILS;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.InterpreterDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+import org.apache.commons.validator.routines.EmailValidator;
+
+@Component
+public class UpdateInterpreterDetailsMidEventHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private static final String INVALID_EMAIL_ERROR_MESSAGE = "Interpreter %s email address is invalid, please enter a valid email address.";
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.MID_EVENT
+            && callback.getEvent() == Event.UPDATE_INTERPRETER_DETAILS;
+    }
+
+    /**
+     * This handler is used to validate the interpreter details email address entered by the user.
+     */
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage,
+                                                        Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+        Optional<List<IdValue<InterpreterDetails>>> optionalInterpreterDetailsList = asylumCase.read(INTERPRETER_DETAILS);
+
+        if (optionalInterpreterDetailsList.isPresent()) {
+            validateInterpretersEmailAddresses(optionalInterpreterDetailsList, response);
```

Comment:
I think if you make the field of type "Email" rather than "Text" in the definitions, CCD will take care of it automatically without the need for the backend to do the validation

Reply:
Ah I see, think I might have seen them before but just forgot about them over time. I have removed my validation and use the one provided by CCD instead, as there are no specific validation requirements in the story yet.